worker/src/enclave/api.rs

```diff
@@ -26,89 +26,15 @@ use sgx_urts::SgxEnclave;
 
 use codec::{Decode, Encode};
 use my_node_runtime::Header;
-use substratee_node_primitives::SignedBlock;
 use sp_core::ed25519;
 use sp_finality_grandpa::VersionedAuthorityList;
+use substratee_enclave_api::enclave_api::EnclaveApi;
+use substratee_enclave_api::error::Error as EnclaveApiError;
+use substratee_enclave_api::{Enclave, EnclaveResult};
+use substratee_node_primitives::SignedBlock;
+use substratee_settings::files::{ENCLAVE_FILE, ENCLAVE_TOKEN};
 
-use substratee_settings::{
-    worker::{EXTRINSIC_MAX_SIZE, STATE_VALUE_MAX_SIZE},
-    files::{ENCLAVE_FILE, ENCLAVE_TOKEN}
-};
-
-extern "C" {
-    fn init(eid: sgx_enclave_id_t, retval: *mut sgx_status_t) -> sgx_status_t;
-
-    fn get_state(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        cyphertext: *const u8,
-        cyphertext_size: u32,
-        shard: *const u8,
-        shard_size: u32,
-        value: *mut u8,
-        value_size: u32,
-    ) -> sgx_status_t;
-
-    fn init_chain_relay(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        genesis_hash: *const u8,
-        genesis_hash_size: usize,
-        authority_list: *const u8,
-        authority_list_size: usize,
-        authority_proof: *const u8,
-        authority_proof_size: usize,
-        latest_header: *mut u8,
-        latest_header_size: usize,
-    ) -> sgx_status_t;
-
-    fn produce_blocks(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        blocks: *const u8,
-        blocks_size: usize,
-        nonce: *const u32,
-    ) -> sgx_status_t;
-
-    fn get_rsa_encryption_pubkey(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        pubkey: *mut u8,
-        pubkey_size: u32,
-    ) -> sgx_status_t;
-
-    fn get_ecc_signing_pubkey(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        pubkey: *mut u8,
-        pubkey_size: u32,
-    ) -> sgx_status_t;
-
-    fn get_mrenclave(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        mrenclave: *mut u8,
-        mrenclave_size: u32,
-    ) -> sgx_status_t;
-
-    fn perform_ra(
-        eid: sgx_enclave_id_t,
-        retval: *mut sgx_status_t,
-        genesis_hash: *const u8,
-        genesis_hash_size: u32,
-        nonce: *const u32,
-        w_url: *const u8,
-        w_url_size: u32,
-        unchecked_extrinsic: *mut u8,
-        unchecked_extrinsic_size: u32,
-    ) -> sgx_status_t;
-
-    fn dump_ra_to_disk(eid: sgx_enclave_id_t, retval: *mut sgx_status_t) -> sgx_status_t;
-
-    fn test_main_entrance(eid: sgx_enclave_id_t, retval: *mut sgx_status_t) -> sgx_status_t;
-}
-
-pub fn enclave_init() -> SgxResult<SgxEnclave> {
+pub fn enclave_init() -> EnclaveResult<Enclave> {
```

Comment:
This function creates and initializes the `SgxEnclave` and its wrapper, the `Enclave` structs. It's called once at the startup of the enclave now (not separately for every command in the worker, where it ended up being called multiple times for certain commands)